File: Integrations/sdks/python/scriptbee_analysis_sdk/abstractions/run_analysis_use_case.py
from abc import ABC, abstractmethod

from scriptbee_analysis_sdk.abstractions.run_analysis_command import RunAnalysisCommand
from scriptbee_analysis_sdk.domain.analysis import AnalysisInfo


class RunAnalysisUseCase(ABC):
    @abstractmethod
    async def run(self, command: RunAnalysisCommand) -> AnalysisInfo: ...

File: Integrations/sdks/typescript/src/abstractions/run-analysis-use-case.ts
import type { ProjectId, ScriptId } from "../domain/project.js";
import type { AnalysisInfo } from "../domain/analysis.js";

export interface RunAnalysisCommand {
  readonly projectId: ProjectId;
  readonly scriptId: ScriptId;
}

export interface RunAnalysisUseCase {
  run(command: RunAnalysisCommand): Promise<AnalysisInfo>;
}
